### PR TITLE
update to nan2

### PR DESCRIPTION
**Pull request by sam-github**
connected to strongloop-internal/scrum-nodeops#893

### CODE REVIEW

**Comment by rmg** [comment]
If #4 can get merged before this is will make CI useable for this PR

**Comment by rmg** [comment]
@slnode test please

File: core.cc [hunk 1]
```diff
@@ -43,17 +43,16 @@ static char ident[1024];
 
 // wrap: void openlog(const char *ident, int option, int facility);
 NAN_METHOD(OpenLog) {
-    NanScope();
```
Comment (sam-github) [inline core.cc]: is this implicit in NAN_METHOD now?

**Comment by sam-github** [comment]
are you sure about the removal of per/method scopes? Other than that, LGTM

File: core.cc (hunk 1 above)
Comment (rmg) [inline core.cc]: Actually I'm not sure. This was done by the `1to2` tool. I'll double check, though.

Comment (rmg) [inline core.cc]: https://github.com/nodejs/nan/blob/master/doc/scopes.md#nanhandlescope

> Note that an implicit `HandleScope` is created for you on JavaScript-accessible methods so you do not need to insert one yourself.

**Comment by rmg** [comment]
@bnoordhuis are you able to give this a once over? You're far more familiar with this stuff than I am and I'd like a second opinion on the scope question.

File: core.cc (hunk 1 above)
Comment (sam-github) [inline core.cc]: ok, looks good then.

File: core.cc [hunk 2]
```diff
@@ -43,17 +43,16 @@ static char ident[1024];
...
 
     // openlog requires ident be statically allocated. Write doesn't guarantee
     // NULL-termination, so preserve last byte as NULL.
-    args[0]->ToString()->WriteUtf8(ident, sizeof(ident)-1);
-    int option = args[1]->ToInt32()->Value();
-    int facility = args[2]->ToInt32()->Value();
+    info[0]->ToString()->WriteUtf8(ident, sizeof(ident)-1);
+    int option = info[1]->ToInt32()->Value();
+    int facility = info[2]->ToInt32()->Value();
```
Comment (bnoordhuis) [inline core.cc]: You can use Int32Value() here.

File: core.cc [hunk 3]
```diff
@@ -78,61 +77,54 @@ static char* dupStr(const Local<String>& m) {
 
 // wrap: void syslog(int priority, const char *format, ...);
 NAN_METHOD(SysLog) {
-    NanScope();
 
-    int priority = args[0]->ToInt32()->Value();
+    int priority = info[0]->ToInt32()->Value();
```
Comment (bnoordhuis) [inline core.cc]: Ditto.

**Comment by bnoordhuis** [comment]
Suggestions but basically LGTM.

**Comment by rmg** [comment]
@slnode test please

**Pull request merged**